server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -816,7 +826,10 @@ private void setRetentionLeaseMillis(final TimeValue retentionLease) {
      * Specialized merge-on-flush policy if provided
      */
     private volatile UnaryOperator<MergePolicy> mergeOnFlushPolicy;
-
+    /**
+     * Is flush check by write threads enabled or not
+     */
+    private volatile boolean checkPendingFlushEnabled;
```

Comment:
If this is not updatable, can this be final?

Reply:
Done